keras/applications/mobilenet.py

```diff
@@ -0,0 +1,616 @@
+        warnings.warn('The MobileNet family of models is only available '
+                      'for the input data format "channels_last" '
+                      '(width, height, channels). '
+                      'However your settings specify the default '
+                      'data format "channels_first" (channels, width, height). '
+                      'You should set `image_data_format="channels_last"` in your Keras '
+                      'config located at ~/.keras/keras.json. '
+                      'The model being returned right now will expect inputs '
+                      'to follow the "channels_last" data format.')
+        K.set_image_data_format('channels_last')
+        old_data_format = 'channels_first'
+    else:
+        old_data_format = None
+
+    # Determine proper input shape. Note, include_top is False by default, as
+    # input shape can be anything larger than 32x32 and the same number of parameters
+    # will be used.
+    input_shape = _obtain_input_shape(input_shape,
+                                      default_size=224,
+                                      min_size=32,
+                                      data_format=K.image_data_format(),
+                                      include_top=False)
+
+    if input_tensor is None:
+        img_input = Input(shape=input_shape)
+    else:
+        if not K.is_keras_tensor(input_tensor):
+            img_input = Input(tensor=input_tensor, shape=input_shape)
+        else:
+            img_input = input_tensor
+
+    x = _conv_block(img_input, 32, alpha, strides=(2, 2))
+    x = _depthwise_conv_block(x, 64, alpha, depth_multiplier, id=1)
+
+    x = _depthwise_conv_block(x, 128, alpha, depth_multiplier, strides=(2, 2), id=2)
+    x = _depthwise_conv_block(x, 128, alpha, depth_multiplier, id=3)
+
+    x = _depthwise_conv_block(x, 256, alpha, depth_multiplier, strides=(2, 2), id=4)
+    x = _depthwise_conv_block(x, 256, alpha, depth_multiplier, id=5)
+
+    x = _depthwise_conv_block(x, 512, alpha, depth_multiplier, strides=(2, 2), id=6)
+    x = _depthwise_conv_block(x, 512, alpha, depth_multiplier, id=7)
+    x = _depthwise_conv_block(x, 512, alpha, depth_multiplier, id=8)
+    x = _depthwise_conv_block(x, 512, alpha, depth_multiplier, id=9)
+    x = _depthwise_conv_block(x, 512, alpha, depth_multiplier, id=10)
+    x = _depthwise_conv_block(x, 512, alpha, depth_multiplier, id=11)
+
+    x = _depthwise_conv_block(x, 1024, alpha, depth_multiplier, strides=(2, 2), id=12)
+    x = _depthwise_conv_block(x, 1024, alpha, depth_multiplier, id=13)
+
+    if include_top:
+        if K.image_data_format() == 'channels_first':
+            shape = (int(1024 * alpha), 1, 1)
+        else:
+            shape = (1, 1, int(1024 * alpha))
+
+        x = GlobalAveragePooling2D()(x)
+        x = Reshape(shape, name='reshape_1')(x)
+        x = Dropout(dropout, name='dropout')(x)
+        x = Convolution2D(classes, (1, 1), padding='same', name='conv_preds')(x)
```

Comment:
@titu1994 I was reading the MobileNets paper, and I was curious why you decided to use a Conv2D layer here instead of a Dense layer, since this layer is supposed to be the fully-connected one.

I know in the end the weights will make the Conv2D layer equivalent to a fully-connected one, but is there any particular reason you used Conv2D here instead of Dense?  Just wanted to better improve my understanding, thanks so much!